### PR TITLE
[Pagination] Add pagination component

### PR DESCRIPTION
**Pull request by astrauka**
### What changed

Added Pagination component

### Why it changed

There was no such component

---

- [x] Change is tested
- [ ] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

**Comment by astrauka** [comment]
@hepiyellow please assist me on
> Keyboard focus outline should appear only when button is accessed with a keyboard. Now it appears also on a mouse click.
For a component coming from wix ui core.

File: src/Pagination/Pagination.st.css [hunk 1]
```diff
@@ -0,0 +1,61 @@
+:import {
+  -st-from: "wix-ui-core/dist/src/components/Pagination/Pagination.st.css";
+  -st-default: Pagination;
+}
+
+:import {
+  -st-from: "wix-ui-backoffice/dist/src/typography.st.css";
```
Comment (shlomitc) [inline src/Pagination/Pagination.st.css]: typography should be taken `Text/Text.st.css`

File: src/Pagination/Pagination.st.css [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  -st-named: fontLight;
+}
+
+:import {
+  -st-from: "wix-ui-backoffice/dist/src/colors.st.css";
```
Comment (shlomitc) [inline src/Pagination/Pagination.st.css]: please create `src/colors.st.css` - it should be the same as `src/colors.scss` only using the `stylable` syntax.
you can add `WHITE` for both `scss` and `st.css`

File: stories/Pagination/index.story.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    currentPage: 8,
+    maxPagesToShow: 9,
+    showFirstPage: true,
+    showLastPage: true,
+    responsive: false,
+    showFirstLastNavButtons: false,
+    paginationMode: 'pages',
+    showInputModeTotalPages: false,
+    disabled: false,
+    previousLabel: <ChevronLeft/>,
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: I don't have the spec for it, but I guess this should be part of the `defaultProps` of the component

File: stories/Pagination/index.story.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    nextLabel: <ChevronRight/>
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: same as `previousLabel`

File: stories/Pagination/index.story.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import {storySettings} from './storySettings';
+
+export default {
+  category: storySettings.kind,
+  storyName: storySettings.story,
+  component: Pagination,
+  componentPath: '../../src/Pagination',
+
+  componentProps: {
+    'data-hook': storySettings.dataHook,
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: `dataHook` to stay consistent with the rest of the components

File: stories/Pagination/index.story.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    totalPages: 15,
+    currentPage: 8,
+    maxPagesToShow: 9,
+    showFirstPage: true,
+    showLastPage: true,
+    responsive: false,
+    showFirstLastNavButtons: false,
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: defaultProps?

File: stories/Pagination/index.story.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    paginationMode: 'pages',
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: defaultProps?

File: stories/Pagination/index.story.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  componentPath: '../../src/Pagination',
+
+  componentProps: {
+    'data-hook': storySettings.dataHook,
+    totalPages: 15,
+    currentPage: 8,
+    maxPagesToShow: 9,
+    showFirstPage: true,
+    showLastPage: true,
+    responsive: false,
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: defaultProps?

File: stories/Pagination/index.story.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  component: Pagination,
+  componentPath: '../../src/Pagination',
+
+  componentProps: {
+    'data-hook': storySettings.dataHook,
+    totalPages: 15,
+    currentPage: 8,
+    maxPagesToShow: 9,
+    showFirstPage: true,
+    showLastPage: true,
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: defaultProps?

File: stories/Pagination/index.story.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  storyName: storySettings.story,
+  component: Pagination,
+  componentPath: '../../src/Pagination',
+
+  componentProps: {
+    'data-hook': storySettings.dataHook,
+    totalPages: 15,
+    currentPage: 8,
+    maxPagesToShow: 9,
+    showFirstPage: true,
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: defaultProps?

File: stories/Pagination/index.story.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    showLastPage: true,
+    responsive: false,
+    showFirstLastNavButtons: false,
+    paginationMode: 'pages',
+    showInputModeTotalPages: false,
```
Comment (shlomitc) [inline stories/Pagination/index.story.js]: defaultProps?

File: src/Pagination/Pagination.st.css (hunk 1 above)
Comment (milkyfruit) [reply src/Pagination/Pagination.st.css]: @shlomitc pagination component was design in Wix blog team, so currently the specs are only in their project: zpl.io/bAEr6v8

Comment (shlomitc) [reply src/Pagination/Pagination.st.css]: @milkyfruit 
so let's add it to our zeplin as a work-in-progress.
I need it to be visible so we can review and discuss it  there as well

**Comment by astrauka** [comment]
@shlomitc moved the component to WIP, could we merge this PR after I resolve the conflicts and then somebody will be able to continue working on the component later?

**Comment by shlomitc** [comment]
@astrauka - sure. Please only fix the conflicting file and a small issue:
I see that the *Misc. Props* section in the story is not working. I didn't investigate it, only saw an error in the console. maybe mandatory props are missing?
Also when clicking on an option - an error is thrown.

@milkyfruit - please approve/request for changes

**Comment by shlomitc** [comment]
For the different keyboard/mouse focus cases we use the `Focusable` hoc.
https://github.com/wix/wix-ui/tree/master/packages/wix-ui-core/src/hocs/Focusable

It might require a change to the Pagination component in `wix-ui-core` though.
@astrauka - if the implementation is not straightforward and you encounter issues, please let me know. In such case we will merge it and handle it later

**Comment by astrauka** [comment]
> For the different keyboard/mouse focus cases we use the Focusable hoc.

@shlomitc already tried to hook it up with Arijus, could not get it right. Need your assistance.

**Comment by shlomitc** [comment]
I'm closing the PR as this component in stale state. no design system definition nor active work on it.
I opened a backlog issue to work on this component once specs are ready

**Comment by argshook** [comment]
fyi, Karolis is on long vacation (for like 6 months), so it's gone stale

**Pull request closed without merging**